### PR TITLE
Add telemetry events

### PR DESCRIPTION
**Pull request by npuvvada**
## Why is this change being made?
Present Telemetry is broken and doesn't generate events. Also, it is limited and not captures all fixups and features.

## What changed?
More tracelogging events are added to capture telemetry across all fixups and features. 

## How was the change tested?
Event generation is verified through TRTT

### CODE REVIEW

File: PsfRuntime/Config.cpp [hunk 1]
```diff
@@ -578,11 +622,24 @@ PSFAPI const psf::json_object* __stdcall PSFQueryExeConfig(const wchar_t* execut
...
+        "Exceptions",
+        TraceLoggingWideString(L"PSFConfigException", "Type"),
+        TraceLoggingWideString((L"No matching exe config found for " + std::wstring(executable)).c_str(), "Message"),
+        TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage),
+        TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES));
     return nullptr;
 }
 catch (...)
 {
+    TraceLoggingWrite(
```
Comment (rahul89dadhich) [inline PsfRuntime/Config.cpp:636]: Can we have difference message to identify the difference between both ?

File: fixups/DynamicLibraryFixup/DynamicLibraryFixup.cpp [hunk 2]
```diff
@@ -104,6 +116,14 @@ HMODULE __stdcall LoadLibraryExFixup(_In_ const CharT* libFileName, _Reserved_ H
                 }
                 catch (...)
                 {
+                    TraceLoggingWrite(g_Log_ETW_ComponentProvider, // handle to my provider
+                        "Exceptions",
```
Comment (rahul89dadhich) [inline fixups/DynamicLibraryFixup/DynamicLibraryFixup.cpp:120]: why we are having same logging messages?

File: fixups/EnvVarFixup/EnvVarFixup.cpp [hunk 3]
```diff
@@ -311,6 +322,14 @@ BOOL __stdcall SetEnvironmentVariableFixup(_In_ const CharT* lpName, _In_ const
             }
             catch (...)
             {
+                TraceLoggingWrite(g_Log_ETW_ComponentProvider, // handle to my provider
+                    "Exceptions",
```
Comment (rahul89dadhich) [inline fixups/EnvVarFixup/EnvVarFixup.cpp:326]: Same message content

File: fixups/RegLegacyFixups/RegistryFixups.cpp [hunk 4]
```diff
@@ -230,6 +242,14 @@ REGSAM RegFixupSam(std::string keypath, REGSAM samDesired, DWORD RegLocalInstanc
                             }
                             catch (...)
                             {
+                                TraceLoggingWrite(g_Log_ETW_ComponentProvider, // handle to my provider
+                                    "Exceptions",
```
Comment (rahul89dadhich) [inline fixups/RegLegacyFixups/RegistryFixups.cpp:246]: same as previous, please check fix all, if required

**Review by rahul89dadhich — CHANGES_REQUESTED** [review]
please check comments

File: include/psf_constants.h [hunk 5]
```diff
@@ -19,6 +19,8 @@ namespace psf
 
     constexpr char arch_string[] = "32";
     constexpr wchar_t warch_string[] = L"32";
+
+    constexpr wchar_t current_version[] = L"1.0.221230.1";
```
Comment (dhoehna) [inline include/psf_constants.h:23]: Big possibility for this number to not be updated during a new release.  Connect with the Windows App SDK folks to figure out how they managed this.

My suggestion is to change this number to a constant local copy.  Something like 0.0.05.  Then, during the official build, replace this value with the correct version.

File: PsfLauncher/PsfPowershellScriptRunner.h [hunk 6]
```diff
@@ -157,6 +160,21 @@ class PsfPowershellScriptRunner
 			std::wstring exitCodeStr = std::to_wstring(exitCode);
 			MessageBoxEx(NULL, (script.scriptPath + std::wstring(L" execution failed with Exit Code ") + exitCodeStr + std::wstring(L". To fix this, please run ") + script.scriptPath + std::wstring(L" standalone in a PowerShell window and confirm that the value of $LASTEXITCODE is 0 (Success) before using it with PSF.")).c_str(), L"Package Support Framework", MB_OK | MB_ICONWARNING, 0);
 		}
+
+		TraceLoggingWrite(
+			g_Log_ETW_ComponentProvider,
+			"ScriptInformation",
+			TraceLoggingWideString((&script == &(this->m_startingScriptInformation)) ? L"StartScript" : L"EndScript", "ScriptType"),
```
Comment (dhoehna) [inline PsfLauncher/PsfPowershellScriptRunner.h:167]: I don't see any documentation about adding new TraceLogging calls to new fixups.  I suggest to add documentation on where, and how, to add traces.

**Review by rahul89dadhich — APPROVED** [review]
looks good

**Comment by dhoehna** [comment]
Looks good. :)

**Pull request merged**